spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java

```diff
@@ -0,0 +1,233 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.spark.connector;
+
+import com.datastrato.gravitino.dto.rel.partitioning.Partitioning.SingleFieldPartitioning;
+import com.datastrato.gravitino.rel.expressions.Expression;
+import com.datastrato.gravitino.rel.expressions.NamedReference;
+import com.datastrato.gravitino.rel.expressions.distributions.Distribution;
+import com.datastrato.gravitino.rel.expressions.distributions.Distributions;
+import com.datastrato.gravitino.rel.expressions.sorts.SortOrder;
+import com.datastrato.gravitino.rel.expressions.sorts.SortOrders;
+import com.datastrato.gravitino.rel.expressions.transforms.Transform;
+import com.datastrato.gravitino.rel.expressions.transforms.Transforms;
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.ws.rs.NotSupportedException;
+import lombok.Getter;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.spark.sql.connector.expressions.BucketTransform;
+import org.apache.spark.sql.connector.expressions.Expressions;
+import org.apache.spark.sql.connector.expressions.IdentityTransform;
+import org.apache.spark.sql.connector.expressions.LogicalExpressions;
+import org.apache.spark.sql.connector.expressions.SortedBucketTransform;
+import scala.collection.JavaConverters;
+
+/**
+ * SparkTransformConverter translate between Spark transform and Gravitino partition, distribution,
+ * sort orders. There may be multi partition transforms, but should be only one bucket transform.
+ * Spark bucket transform is corresponding to Gravitino Hash distribution without sort orders. Spark
+ * sorted bucket transform is corresponding to Gravitino Hash distribution with sort orders.
+ */
+public class SparkTransformConverter {
+
+  public static class GravitinoTransformBundles {
+    private List<Transform> partitions;
+    @Getter private Distribution distribution;
+    @Getter private SortOrder[] sortOrders;
+
+    public Transform[] getPartitions() {
+      if (partitions == null) {
+        return null;
+      }
+      return partitions.toArray(new Transform[0]);
+    }
+
+    private void addPartition(Transform partition) {
+      if (partitions == null) {
+        this.partitions = new ArrayList<>();
+      }
+      partitions.add(partition);
+    }
+
+    private void setDistribution(Distribution distribution) {
+      Preconditions.checkState(this.distribution == null, "Should only set distribution once");
```

Comment:
remove "this."

Reply:
`this. distribution` is used to distinguish with `distribution` parameter, use a different name for method parameter or keep use `this. distribution`, which do you prefer?

Reply:
That's the ambiguity / potential bug you introduced, you should fix that issue, not using "this.xxx".

Reply:
done